src/engine/enginepregain.cpp

```diff
@@ -51,36 +53,46 @@ EnginePregain::EnginePregain(QString group)
 EnginePregain::~EnginePregain()
 {
     delete m_pPotmeterPregain;
-    delete m_pControlReplayGain;
+    delete m_pCOReplayGain;
     delete m_pTotalGain;
 
     delete s_pEnableReplayGain;
     s_pEnableReplayGain = NULL;
     delete s_pReplayGainBoost;
     s_pReplayGainBoost = NULL;
+    delete s_pDefaultBoost;
+    s_pDefaultBoost = NULL;
 }
 
 void EnginePregain::process(CSAMPLE* pInOut, const int iBufferSize) {
-
-    float fEnableReplayGain = s_pEnableReplayGain->get();
-    const float fReplayGainBoost = s_pReplayGainBoost->get();
-    float fGain = m_pPotmeterPregain->get();
-    float fReplayGain = m_pControlReplayGain->get();
-    float fReplayGainCorrection = 1;
-    float fPassing = m_pPassthroughEnabled->get();
-
-    // Override replaygain value if passing through
-    if (fPassing > 0) {
-        fReplayGain = 1.0;
-    } else if (fReplayGain * fEnableReplayGain != 0) {
-        // Here is the point, when ReplayGain Analyser takes its action, suggested gain changes from 0 to a nonzero value
+    const float fReplayGain = m_pCOReplayGain->get();
+    float fReplayGainCorrection;
+    if (m_pPassthroughEnabled->toBool()) {
+        // Override replaygain value if passing through
+        // TODO(XXX): consider a good default.
+        // Do we expect an replaygain leveled input or
+        // Normalized to 1 input?
+        fReplayGainCorrection = 1; // We expect a replaygain leveled input
+    } else if (!s_pEnableReplayGain->toBool() || fReplayGain == 0) {
```

Comment:
@rryan: The current version works for me, but since I do not like to merge this if you have still objections, I need to know what I can change that it works for you as well.  Pleas see my comments above. 

Do we need an addition check box that switched the boost slider?